File: nesoy.kwon/src/main/java/homework/chapter_6/Team.java
package homework.chapter_6;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table
public class Team {

	@Id
	@Column(name = "TEAM_ID")
	private String id;

	private String name;

	@OneToMany
	private List<Member> members = new ArrayList<>();

	public List<Member> getMembers() {
		return members;
	}

	public void setMembers(List<Member> members) {
		this.members = members;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	@Override
	public String toString() {
		return "Team{" +
				"id='" + id + '\'' +
				", name='" + name + '\'' +
				", members.size()=" + members.size() +
				'}';
	}
}

File: hyun.k/hyun.k/src/main/kotlin/chapter6/homework1/Team.kt
package chapter6.homework1

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToMany

@Entity(name = "team")
data class Team(

    @Id
    @Column
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column
    val name: String? = null,

    @OneToMany
    @JoinColumn(name = "team_id")
    val members: MutableList<Member> = mutableListOf(),
)
